pkg/service/backup/restore_worker_data.go

```diff
@@ -25,22 +25,30 @@ func (w *restoreWorker) restoreData(ctx context.Context, run *RestoreRun, target
 		w.hosts = nil
 
 		manifestHandler := func(miwc ManifestInfoWithContent) error {
-			w.miwc = miwc
-			w.Logger.Info(ctx, "Restoring manifest", "manifest", miwc.ManifestInfo)
 			// Check if manifest has already been processed in previous run
 			if !w.resumed && run.ManifestPath != miwc.Path() {
 				return nil
 			}
+
+			w.Logger.Info(ctx, "Restoring manifest", "manifest", miwc.ManifestInfo)
+			defer w.Logger.Info(ctx, "Restoring manifest finished", "manifest", miwc.ManifestInfo)
+
+			w.miwc = miwc
 			run.ManifestPath = miwc.Path()
 
 			return miwc.ForEachIndexIterWithError(target.Keyspace, w.restoreFiles(ctx, run, target))
 		}
 
 		if err := w.forEachRestoredManifest(ctx, w.location, manifestHandler); err != nil {
+			w.Logger.Info(ctx, "Restoring location finished", "location", w.location)
 			return err
 		}
+
+		w.Logger.Info(ctx, "Restoring location finished", "location", w.location)
```

Comment:
Maybe `Restoring location completed` would be better ?
Completed means fulfilled, finished means stopped with no matter what result.

https://ejoy-english.com/blog/complete-vs-finish-similar-but-different/#:~:text='To%20complete%20something'%20means%20to,something%20means%20to%20end%20it.

Reply:
Answered to that in above comment.

Reply:
Ok